sim/hunter/talents.go

```diff
@@ -1,870 +1,823 @@
-	// }
-	// if hunter.Talents.RangedWeaponSpecialization > 0 {
-	// 	mult := 1 + []float64{0, .01, .03, .05}[hunter.Talents.RangedWeaponSpecialization]
-	// 	hunter.OnSpellRegistered(func(spell *core.Spell) {
-	// 		if spell.ProcMask.Matches(core.ProcMaskRanged) {
-	// 			spell.DamageMultiplier *= mult
-	// 		}
-	// 	})
-	// }
-
-	// if hunter.Talents.EnduranceTraining > 0 {
-	// 	healthBonus := 0.01 * float64(hunter.Talents.EnduranceTraining)
-	// 	hunter.MultiplyStat(stats.Health, 1.0+healthBonus)
-	// 	if hunter.pet != nil {
-	// 		hunter.pet.MultiplyStat(stats.Health, 1.0+2*healthBonus)
-	// 	}
-	// }
-
-	// if hunter.Talents.ThickHide > 0 {
-	// 	var hunterBonus, petBonus float64
-	// 	if hunter.Talents.ThickHide == 1 {
-	// 		hunterBonus = 0.04
-	// 		petBonus = 0.07
-	// 	} else if hunter.Talents.ThickHide == 2 {
-	// 		hunterBonus = 0.07
-	// 		petBonus = 0.14
-	// 	} else if hunter.Talents.ThickHide == 3 {
-	// 		hunterBonus = 0.1
-	// 		petBonus = 0.2
-	// 	}
-	// 	hunter.ApplyEquipScaling(stats.Armor, 1.0+hunterBonus)
-	// 	if hunter.pet != nil {
-	// 		hunter.pet.MultiplyStat(stats.Armor, 1.0+petBonus)
-	// 	}
-	// }
-
-	// if hunter.Talents.Survivalist > 0 {
-	// 	hunter.MultiplyStat(stats.Stamina, 1.0+0.02*float64(hunter.Talents.Survivalist))
-	// }
-
-	// if hunter.Talents.CombatExperience > 0 {
-	// 	bonus := 1.0 + (0.02 * float64(hunter.Talents.CombatExperience))
-	// 	hunter.MultiplyStat(stats.Agility, bonus)
-	// 	hunter.MultiplyStat(stats.Intellect, bonus)
-	// }
-	// if hunter.Talents.CarefulAim > 0 {
-	// 	hunter.AddStatDependency(stats.Intellect, stats.RangedAttackPower, (1.0/3.0)*float64(hunter.Talents.CarefulAim))
-	// }
-	// if hunter.Talents.HunterVsWild > 0 {
-	// 	bonus := 0.1 * float64(hunter.Talents.HunterVsWild)
-	// 	hunter.AddStatDependency(stats.Stamina, stats.AttackPower, bonus)
-	// 	hunter.AddStatDependency(stats.Stamina, stats.RangedAttackPower, bonus)
-	// }
-	// if hunter.Talents.LightningReflexes > 0 {
-	// 	agiBonus := 0.03 * float64(hunter.Talents.LightningReflexes)
-	// 	hunter.MultiplyStat(stats.Agility, 1.0+agiBonus)
-	// }
-	// if hunter.Talents.HuntingParty > 0 {
-	// 	agiBonus := 0.01 * float64(hunter.Talents.HuntingParty)
-	// 	hunter.MultiplyStat(stats.Agility, 1.0+agiBonus)
-	// }
-
-	// hunter.applySpiritBond()
-	// hunter.applyInvigoration()
-	// hunter.applyCobraStrikes()
-	// hunter.applyGoForTheThroat()
-	// hunter.applyPiercingShots()
-	// hunter.applyWildQuiver()
-	// hunter.applyImprovedTracking()
-	// hunter.applyThrillOfTheHunt()
-	// hunter.applyLockAndLoad()
-	// hunter.applyExposeWeakness()
-	// hunter.applyMasterTactician()
-	// hunter.applySniperTraining()
-	// hunter.applyHuntingParty()
-
-	// hunter.registerReadinessCD()
+	if hunter.Pet != nil {
+		hunter.applyFrenzy()
+		hunter.registerBestialWrathCD()
+		// Todo: BM stuff
+		hunter.Pet.ApplyTalents()
+	}
+
+	if hunter.Talents.Pathing > 0 {
+		bonus := 0.03 * float64(hunter.Talents.Pathing)
+		hunter.MultiplyCastSpeed(bonus)
```

Comment:
Cant use those functions in talent because they modify swing timers no? I looked for them, MultiplyCastSpeed was the most reasonable one I could find. I'll do `unit.PseudoStats.RangedSpeedMultiplier *= amount` instead but again, dunno if the right value is there when im trying to set it.

Reply:
yes, if this is done during initialization and not dyanimacly then its the pseudo stat you have mentioned didnt realize this is not dynamic aura bonus